### PR TITLE
Allow wildcard for view instrument name

### PR DESCRIPTION
**Pull request by albertored**
Complete the implementation of wildcard views

### CODE REVIEW

**Comment by albertored** [comment]
@tsloughter by working on this I notice that the `otel_view` module contains a lot of dialyzer ignores due to the fact that we are using the instrument record in match specs but its type is strictly defined. Now these ignores are starting to percolate outside that module so maybe it's time to extend the instrument type with match specs as already done for [aggreagators](https://github.com/open-telemetry/opentelemetry-erlang/blob/main/apps/opentelemetry_experimental/include/otel_metrics.hrl#L5). The only other solution (beside keeping the ignores) is to use the raw tuple instead of the record when used in match specs, Wat do you think?

**Review by tsloughter — APPROVED** [review]

**Comment by tsloughter** [comment]
@albertored oh yea, and definitely time to extend the type.

**Comment by albertored** [comment]
@tsloughter perfect, I'll add it to this PR if it's ok

**Comment by albertored** [comment]
@tsloughter done, but there is still a failing test

**Comment by albertored** [comment]
@tsloughter rebased to fix conflicts

**Review by tsloughter — APPROVED** [review]

**Comment by tsloughter** [comment]
@albertored looks good. rebase on latest `main` and I'll merge

File: apps/opentelemetry_api_experimental/include/otel_metrics.hrl [hunk 1]
```diff
@@ -1,12 +1,16 @@
--record(instrument, {module        :: module(),
-                     meter         :: otel_meter:t(),
-                     name          :: otel_instrument:name(),
-                     description   :: otel_instrument:description() | undefined,
-                     kind          :: otel_instrument:kind(),
-                     unit          :: otel_instrument:unit() | undefined,
-                     temporality   :: otel_instrument:temporality(),
-                     callback      :: otel_instrument:callback() | undefined,
-                     callback_args :: otel_instrument:callback_args() | undefined}).
+-ifndef(MATCH_SPEC_TYPES_DEFINED).
```
[the remark this answers is not in the document]
Comment (albertored) [reply apps/opentelemetry_api_experimental/include/otel_metrics.hrl:1]: Yes, that's smarter, thanks, I'll fix it tomorrow

**Comment by albertored** [comment]
@tsloughter rebased

**Pull request merged**